File: src/copy_trait.ts
class MyPlainStruct {
    foo: number
}

// Note we can only derive `Copy` if we *also* derive `Clone`.
export class MyCopyableStruct {
    constructor(public foo: number) { }
}

export function hurray(): number {
    let a = new MyCopyableStruct(1);
    let b = a;
    return a.foo + b.foo
}

/// # Output
console.log(`🐥 hurray: ${hurray()}`)

// 🐥 hurray: 2

File: src/copy_trait.rs
struct MyPlainStruct {
    foo: u64,
}

// Note we can only derive `Copy` if we *also* derive `Clone`.
#[derive(Clone, Copy, Debug)]
pub struct MyCopyableStruct {
    pub foo: u64, // All primitives are copyable out of the box.
}

pub fn hurray() -> u64 {
    let a = MyCopyableStruct { foo: 1 };
    let b = a;
    a.foo + b.foo
}

/// # Output
#[test]
fn test() {
    println!("🦀 hurray: {}", hurray());
}

// 🦀 hurray: 2
